### DISCUSSION TITLE
Removing execute() function when calling the use case

### DISCUSSION THREAD

**Discussion by lydiasama** [nimblehq/android-templates#257] [Q&A] [answered]
Hi guys 👋🏻 
[@Kaung Khant Soe](https://nimble-co.slack.com/team/U031KRL7ZPA) and I have a topic to discuss with you.
Since Kotlin has this [Overriding invoke method](https://riptutorial.com/kotlin/example/31101/overriding-invoke-method-to-build-dsl), what do you think if we eliminate the execute function of UseCase and use this invoke function instead?
So the UseCase is gonna be like:
```
interface DoSomethingUseCase {
    suspend operator fun invoke(id: String): DoSomethingResult
}
```
Then when we call the UseCase, we can simply call the UseCase without .execute() like:
```
DoSomethingUseCase(id)
```
What do you think if we change the way to implement the use case like this?

**Comment by kaungkhantsoe** [comment]
To make use of invoke() , it make sense like declaring the usecases in the form of `SomeViewModel(val doSomething: DoSomethingUseCase)` instead of `SomeViewModel(val doSomethingUseCase: DoSomethingUseCase)`. So that we can invoke the usecase like `doSomething()` instead of `doSomethingUseCase()`. It is more readable for me. What do you guys think about that?

**Reply by lydiasama** [reply]
IMO, we usually name the argument with the same name as its type, so to have `doSomeThingUseCase()` will make the reader be more clear that it is a use case. 🤔

**Comment by toby-thanathip** [comment, marked as answer]
Issue created: #262

**Reply by lydiasama** [reply]
Will we delete this discussion or leave it as a discussion history? 🤔

**Reply by toby-thanathip** [reply]
Let's leave it as a discussion history 💪
